packages/docs-page/server/loaders/remote-content.ts

```diff
@@ -0,0 +1,197 @@
+}
+
+interface RemoteContentLoaderOpts extends DataLoaderOpts {
+  basePath: string
+  enabledVersionedDocs?: boolean
+}
+
+/**
+ * TODO: export this from future content-api client
+ * @see https://app.asana.com/0/1100423001970639/1201071725174928/f
+ */
+interface VersionMetadataItem {
+  product: string
+  ref: string
+  version: string
+  created_at: string
+  display?: string
+  sha: string
+  isLatest?: boolean
+  fullPath: string
+}
+
+interface VersionSelectItem {
+  name: string
+  label: string
+}
+
+const moizeOpts: Options = { isPromise: true, maxSize: Infinity }
+const cachedFetchNavData = moize(fetchNavData, moizeOpts)
+const cachedFetchVersionMetadataList = moize(
+  fetchVersionMetadataList,
+  moizeOpts
+)
+
+/**
+ * Formats a list of version-metadata to,
+ * be passed to `<VersionSelect versions={[...]} />`.
+ * - Sorts by semver, descending
+ */
+export function mapVersionList(
+  list: VersionMetadataItem[]
+): VersionSelectItem[] {
+  const versions = list
+    .sort((a, b) =>
+      semver.compare(semver.coerce(b.version), semver.coerce(a.version))
+    )
+    .map((e) => {
+      const { isLatest, version, display } = e
+
+      const displayValue = display ?? version
+
+      return {
+        name: isLatest ? 'latest' : version,
+        label: isLatest ? `${displayValue} (latest)` : displayValue,
+      }
+    })
+
+  return versions
+}
+
+export default class RemoteContentLoader implements DataLoader {
+  opts: RemoteContentLoaderOpts
+
+  constructor(opts: RemoteContentLoaderOpts) {
+    this.opts = opts
+
+    if (typeof this.opts.enabledVersionedDocs === 'undefined')
+      this.opts.enabledVersionedDocs =
+        process.env.ENABLE_VERSIONED_DOCS?.toString() === 'true'
+
+    if (!this.opts.paramId) this.opts.paramId = DEFAULT_PARAM_ID
+  }
+
+  loadStaticPaths = async (): Promise<$TSFixMe> => {
+    // Fetch version metadata to get "latest"
+    const versionMetadataList = await cachedFetchVersionMetadataList(
+      this.opts.product
+    )
+    const latest = versionMetadataList.find((e) => e.isLatest).version
+    // Fetch and parse navigation data
+    return getPathsFromNavData(
+      (await cachedFetchNavData(this.opts.product, this.opts.basePath, latest))
+        .navData,
+      this.opts.paramId
+    )
+  }
+
+  loadStaticProps = async ({
+    params,
+    mainBranch = 'main', // we should pull this from config
+    remarkPlugins = [], // do we really need this?
+    scope, // only used once: https://sourcegraph.hashi-mktg.com/search?q=scope:+lang:javascript+file:website*&patternType=literal this can also be passed to the component
+  }: RemoteContentLoaderContext): Promise<$TSFixMe> => {
+    // Build the currentPath from page parameters
+    const currentPath = params[this.opts.paramId]
+      ? params[this.opts.paramId].join('/')
+      : ''
+
+    const mdxRenderer = (mdx) =>
+      renderPageMdx(mdx, {
+        remarkPlugins,
+        scope,
+      })
```

Comment:
Don't know if its actually worth abstracting but I think this exact same thing appears in the fs loader